vendor/istio.io/api/mesh/v1alpha1/config.pb.go

```diff
@@ -354,10 +379,12 @@ type MeshConfig struct {
 	// if sidecar is installed on all pods in the mesh, then this should be set to UPGRADE.
 	// If one or more services or namespaces do not have sidecar(s), then this should be set to DO_NOT_UPGRADE.
 	// It can be enabled by destination using the destinationRule.trafficPolicy.connectionPool.http.h2UpgradePolicy override.
-	H2UpgradePolicy      MeshConfig_H2UpgradePolicy `protobuf:"varint,41,opt,name=h2_upgrade_policy,json=h2UpgradePolicy,proto3,enum=istio.mesh.v1alpha1.MeshConfig_H2UpgradePolicy" json:"h2_upgrade_policy,omitempty"`
-	XXX_NoUnkeyedLiteral struct{}                   `json:"-"`
-	XXX_unrecognized     []byte                     `json:"-"`
-	XXX_sizecache        int32                      `json:"-"`
+	H2UpgradePolicy MeshConfig_H2UpgradePolicy `protobuf:"varint,41,opt,name=h2_upgrade_policy,json=h2UpgradePolicy,proto3,enum=istio.mesh.v1alpha1.MeshConfig_H2UpgradePolicy" json:"h2_upgrade_policy,omitempty"`
+	// Optional: Features is used to enable/disable specific features.
+	Features             []*Feature `protobuf:"bytes,42,rep,name=features,proto3" json:"features,omitempty"`
```

Comment:
@howardjohn Look at here, this is the global api.

Reply:
I see that. I don't think the API is the right choice for this. We had an agenda item to discuss a proper API in the networking wg (as mentioned in the slack discussion that I think you saw).

We already have a locality field in mesh config why was this not added there??

This is no better than the feature flag we have today...

Reply:
I would like to join if possible. But i am afraid of the time difference.

Reply:
https://github.com/istio/istio/pull/15014#discussion_r296589723